server/config/config.go

```diff
@@ -459,12 +518,25 @@ func defineFlagsForMembers(parentStructNames []string, T reflect.Value) {
 	}
 }
 
-// Register flags too.
-func init() {
-	defineFlagsForMembers([]string{}, reflect.ValueOf(&sharedGeneralConfig).Elem())
+func populateYamlPresenceMap(yamlMap map[interface{}]interface{}, presenceMap map[string]struct{}) {
+	var check func(key []string, value interface{})
+	check = func(key []string, value interface{}) {
+		if m, ok := value.(map[interface{}]interface{}); ok {
+			for k, v := range m {
+				if newKey, ok := k.(string); ok {
+					check(append(key, strings.ToLower(newKey)), v)
+					continue
+				}
+				log.Warningf("Non-string key encountered in yaml map in field: %s", key)
+			}
+			return
+		}
+		presenceMap[strings.Join(key, ".")] = struct{}{}
+	}
+	check([]string{}, yamlMap)
 }
 
-func parseConfig(fileBytes []byte) (*generalConfig, error) {
+func parseConfig(fileBytes []byte, presenceMap map[string]struct{}) (*generalConfig, error) {
```

Comment:
Is it necessary to pass the presence map as an out parameter? Could you instantiate the empty map inside the function and return it instead?